onadata/libs/filters.py

```diff
@@ -20,7 +20,8 @@ def filter_queryset(self, request, queryset, view):
         """
         Anonymous user has no object permissions, return queryset as it is.
         """
-        form_id = view.kwargs.get(view.lookup_field)
+        form_id = view.kwargs.get(
+            view.lookup_field, view.kwargs.get('xform_pk'))
```

Comment:
@ivermac yes, if `view.kwargs.get(view.lookup_field)` is `None` we assign `view.kwargs.get('xform_pk')` to `form_id`

Reply:
@ukanga, yes this is/was the issue and the reason the XForm queryset did not include these forms was that this filter returned a 404 [here](https://github.com/onaio/onadata/blob/master/onadata/libs/filters.py#L37) because this filter expected a `kwarg`  called `pk` that is not passed by the viewset.

Reply:
This Fix has been deployed on stage and actually fixes the issue that is detailed [here](https://github.com/onaio/onadata/issues/1753)